### DISCUSSION TITLE
Android OSS app removed?

### DISCUSSION THREAD

**Discussion by ncurzon** [google/mozc#879] [Q&A] [answered]
Hi Google, as a recent user of OSS Android stuff, I had been using Mozc through F-Droid on LineageOS, however, recently I updated to Calyx OS, Android 14, and I can no longer install Mozc because the build on F-Droid is too old.

However, I couldn't find any android app in this repository.

I used bisect to find this commit: https://github.com/google/mozc/commit/a1dcadabd3a1c604e8beff1e45830c1d9adfce84 where a lot of stuff got removed. Was there a reason for this removal, and do you think it would be difficult to fork and get it building again?

**Comment by yukawa** [comment, marked as answer]
> Was there a reason for this removal

Yes. The removal was intentional as we concluded that it’s unrealistic for the Mozc team to keep maintaining the Android client code mainly due to the lack of engineering resources.

> and do you think it would be difficult to fork and get it building again?

Just making it work again probably requires only a few days or weeks of code modernizations, which I’d say would be a “not that difficult” task. There are already several examples in GitHub. You can reach out to these authors to hear their actual experiences and opinions about how difficult it was.

 * https://github.com/elizagamedev/android-libre-japanese-input
 * https://github.com/kachaya/MozcForAndroid
 * https://github.com/Lee0701/mozc-android

Keeping maintaining it and responding to random bug reports and feature requests are basically forever full time jobs. The more your OSS project becomes popular, the more you are requested to do something. Eventually people start expecting production quality or even higher quality especially in the privacy/security areas.
